### PR TITLE
Update Logging for Gradient Accumulation

### PR DESCRIPTION
**Pull request by ashors1**
This PR includes the changes to `train.py` originally from PR https://github.com/google/paxml/pull/8. In particular:

- Fixes logging for gradient accumulation such that reported loss and throughput reflects that of a full batch rather than a subbatch
- Adds `seqs/sec` metric

### CODE REVIEW

File: paxml/programs.py [hunk 1]
```diff
@@ -169,47 +170,55 @@ def setup(
...
+              summary_tensors,
+          ) = self.train_step(
+              state,
+              self._train_prng_seed,
+              model_inputs,
+              self._train_unpadded_global_batch_size,
+          )
+          summed_loss += loss
+          state = new_state
+    del state  # Unused anymore.
```
Comment (zhangqiaorjc) [inline paxml/programs.py:215]: can you indent this to the same level as `state = new_state`

File: paxml/programs.py [hunk 2]
```diff
@@ -316,6 +325,7 @@ def _sync_device(self, new_state: TrainState, train_step: int):
     # a gap between steps.
     new_train_step = int(
         py_utils.maybe_unreplicate_for_fully_replicated(new_state.step)
+        /self._num_sub_batches
```
Comment (zhangqiaorjc) [inline paxml/programs.py:328]: it's unclear to me how this interacts with 

new_train_step = train_step + 1

on line 293

File: paxml/programs.py [hunk 3]
```diff
@@ -169,47 +170,55 @@ def setup(
...
-            summary_tensors,
-        ) = self.train_step(
-            state,
-            self._train_prng_seed,
-            model_inputs,
-            self._train_unpadded_global_batch_size,
-        )
-      del state  # Unused anymore.
+    summed_loss = jnp.zeros((), dtype=jnp.float32)
+    for grad_accum_step in range(self._num_sub_batches):
```
Comment (aaroey) [inline paxml/programs.py:182]: Hi, given most (?) users don't use this feature by default, and changing this loop will affect all users, I think it'll be better to make a subclass of BaseTrainProgram or SingleTaskTrainProgram and add the support there. Can we do that?

We are going to add configuration to support custom training programs in the coming days.
